### PR TITLE
Run3-sim102 Make runtime variables to allow/disallow creation of SimHits for demonstration chambers

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Make runtime variables to allow/disallow creation of SimHits for demonstration chambers

#### PR validation:

Use the runTheMatrix test workflows

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by civanch** [comment]
@bsunanda , what is the reason for lost of regression?

**Comment by bsunanda** [comment]
@civanch I cannot think of the reason for this. With no change of the scenario, I cannot think why regression gives different results. Is it for Run3 or is it for Phase2 scenarios?

**Comment by civanch** [comment]
@bsunanda , regression fail in Phase-2 WFs: 

35034.0 | 717
35234.999 | 1597
38634.0 | 3146

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by civanch** [comment]
+1

File: SimG4Core/Application/python/g4SimHits_cfi.py [hunk 1]
```diff
@@ -669,3 +672,6 @@
 phase2_common.toModify(g4SimHits,
                        OnlySDs = ['ZdcSensitiveDetector', 'TotemT2ScintSensitiveDetector', 'TotemSensitiveDetector', 'RomanPotSensitiveDetector', 'PLTSensitiveDetector', 'MuonSensitiveDetector', 'MtdSensitiveDetector', 'BCM1FSensitiveDetector', 'EcalSensitiveDetector', 'CTPPSSensitiveDetector', 'HGCalSensitiveDetector', 'BSCSensitiveDetector', 'CTPPSDiamondSensitiveDetector', 'FP420SensitiveDetector', 'BHMSensitiveDetector', 'HFNoseSensitiveDetector', 'HGCScintillatorSensitiveDetector', 'CastorSensitiveDetector', 'CaloTrkProcessing', 'HcalSensitiveDetector', 'TkAccumulatingSensitiveDetector'],
                        LHCTransport = False ) 
+phase2_common.toModify( g4SimHits,  MuonSD = dict(
```
Comment (perrotta) [inline SimG4Core/Application/python/g4SimHits_cfi.py:675]: Please unify with the previous `phase2_common.toModify`

File: SimG4CMS/Muon/src/MuonSensitiveDetector.cc [hunk 2]
```diff
@@ -336,3 +353,35 @@ Local3DPoint MuonSensitiveDetector::FinalStepPositionVsParent(const G4Step* curr
...
+    int subdet = DetId(id).subdetId();
+    if (subdet == MuonSubdetId::GEM) {
+      if (GEMDetId(id).station() == 2)
+        flag = demoGEM_;
+    } else if (subdet == MuonSubdetId::RPC) {
+      if ((RPCDetId(id).region() != 0) && (RPCDetId(id).ring() == 1) && (RPCDetId(id).station() > 2))
+        flag = demoRPC_;
+    }
+  }
+  if (id == 0)
```
Comment (perrotta) [inline SimG4CMS/Muon/src/MuonSensitiveDetector.cc:369]: Shouldn't this be checked at the very beginning of the method, and simply return `false` if `id == 0` ?
(Possibly with a debug line attached, if you want so)

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: SimG4CMS/Muon/src/MuonSensitiveDetector.cc (hunk 2 above)
Comment (bsunanda) [reply SimG4CMS/Muon/src/MuonSensitiveDetector.cc]: Taken care of

File: SimG4Core/Application/python/g4SimHits_cfi.py (hunk 1 above)
Comment (bsunanda) [reply SimG4Core/Application/python/g4SimHits_cfi.py]: Done

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by civanch** [comment]
+1

**Comment by perrotta** [comment]
+1

**Pull request merged**